### PR TITLE
Add more tests for Intl.Segmenter

### PR DESCRIPTION
**Pull request by FrankYFTang**
@leobalter @gibson042 @zbraniecki @anba

### CODE REVIEW

File: test/intl402/Segmenter/constructor/constructor/options-granularity-abrupt-throws.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// Copyright 2018 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-intl.segmenter
+description:
+    Return abrupt completion from GetOption localeMatcher
```
Comment (rwaldron) [inline test/intl402/Segmenter/constructor/constructor/options-granularity-abrupt-throws.js:7]: The file name indicates that this should be `granularity`

File: test/intl402/Segmenter/constructor/constructor/options-granularity-abrupt-throws.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    13. Let granularity be ? GetOption(options, "granularity", "string", « "grapheme", "word", "sentence" », "grapheme").
+
+    GetOption ( options, property, type, values, fallback )
+    1. Let value be ? Get(options, property).
+features: [Intl.Segmenter]
+---*/
+
+
+var options = {};
+Object.defineProperty(options, 'localeMatcher', {
```
Comment (rwaldron) [inline test/intl402/Segmenter/constructor/constructor/options-granularity-abrupt-throws.js:20]: Should this be `granularity `?

File: test/intl402/Segmenter/constructor/constructor/options-granularity-toString-abrupt-throws.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+// Copyright 2018 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-intl.segmenter
+description:
+    Return abrupt completion from GetOption localeMatcher
```
Comment (rwaldron) [inline test/intl402/Segmenter/constructor/constructor/options-granularity-toString-abrupt-throws.js:7]: Same question as above

File: test/intl402/Segmenter/constructor/constructor/options-null.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+// Copyright (C) 2019 Leo Balter. All rights reserved.
```
Comment (rwaldron) [inline test/intl402/Segmenter/constructor/constructor/options-null.js:1]: Your name is not Leo Balter ;)

File: test/intl402/Segmenter/constructor/constructor/options-granularity-abrupt-throws.js (hunk 1 above)
Comment (FrankYFTang) [reply test/intl402/Segmenter/constructor/constructor/options-granularity-abrupt-throws.js:7]: > The file name indicates that this should be `granularity`

sorry. My bad. Must be too tired last night and forget to change it after copy the file.

File: test/intl402/Segmenter/constructor/constructor/options-granularity-abrupt-throws.js (hunk 2 above)
Comment (FrankYFTang) [reply test/intl402/Segmenter/constructor/constructor/options-granularity-abrupt-throws.js:20]: > Should this be `granularity `?

sorry. My bad. Must be too tired last night and forget to change it after copy the file.

File: test/intl402/Segmenter/constructor/constructor/options-null.js (hunk 4 above)
Comment (FrankYFTang) [reply test/intl402/Segmenter/constructor/constructor/options-null.js:1]: I copied a file author by him and modify from there. I don't think I should change his original copyright from that, right?

**Comment by FrankYFTang** [comment]
Thanks for catching those issue. PTAL.

**Comment by leobalter** [comment]
From the linter:

```
/home/circleci/test262/test/intl402/Segmenter/prototype/segment/containing/branding.js: INCLUDES - Unused include: "propertyHelper.js"
/home/circleci/test262/test/intl402/Segmenter/prototype/segment/containing/breakable-input.js: INCLUDES - Unused include: "propertyHelper.js"
/home/circleci/test262/test/intl402/Segmenter/prototype/segment/containing/index-throws.js: INCLUDES - Unused include: "propertyHelper.js"
/home/circleci/test262/test/intl402/Segmenter/prototype/segment/containing/iswordlike.js: INCLUDES - Unused include: "propertyHelper.js"
/home/circleci/test262/test/intl402/Segmenter/prototype/segment/containing/one-index.js: INCLUDES - Unused include: "propertyHelper.js"
/home/circleci/test262/test/intl402/Segmenter/prototype/segment/containing/out-of-bound-index.js: INCLUDES - Unused include: "propertyHelper.js"
/home/circleci/test262/test/intl402/Segmenter/prototype/segment/containing/unbreakable-input.js: INCLUDES - Unused include: "propertyHelper.js"
/home/circleci/test262/test/intl402/Segmenter/prototype/segment/containing/word-iswordlike.js: INCLUDES - Unused include: "propertyHelper.js"
/home/circleci/test262/test/intl402/Segmenter/prototype/segment/containing/zero-index.js: INCLUDES - Unused include: "propertyHelper.js"
```

File: test/intl402/Segmenter/prototype/segment/containing/branding.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+/*---
+esid: sec-%segmentsprototype%.containing
+description: Verifies the branding check for the "segment" function of the %Segments.prototype%.containing.
+info: |
+    %Segments.prototype%.containing ( index )
+    Unless specified otherwise in this document, the objects, functions, and constructors described in this standard are subject to the generic requirements and restrictions specified for standard built-in ECMAScript objects in the ECMAScript 2020 Language Specification, 11th edition, clause 17, or successor.
+    Every built-in function object, including constructors, that is not identified as an anonymous function has a name property whose value is a String. Unless otherwise specified, this value is the name that is given to the function in this specification.
+    Unless otherwise specified, the name property of a built-in function object, if it exists, has the attributes { [[Writable]]: false, [[Enumerable]]: false, [[Configurable]]: true }.
+
+includes: [propertyHelper.js]
```
Comment (leobalter) [inline test/intl402/Segmenter/prototype/segment/containing/branding.js:12]:
```suggestion
```

**Review by leobalter — DISMISSED** [review]

File: test/intl402/Segmenter/prototype/segment/containing/branding.js (hunk 5 above)
Comment (leobalter) [reply test/intl402/Segmenter/prototype/segment/containing/branding.js:12]: Same goes for the other files [listed above](https://github.com/tc39/test262/pull/2727#issuecomment-673610872).

Comment (FrankYFTang) [reply test/intl402/Segmenter/prototype/segment/containing/branding.js]: opps.. sorry

**Comment by FrankYFTang** [comment]
PTAL

**Review by rwaldron — DISMISSED** [review]
@leobalter PTAL!

**Comment by FrankYFTang** [comment]
@leobalter Ping

**Comment by FrankYFTang** [comment]
I try to ship Intl.Segmenter for Chrome soon. It will be nice if these tests could be reviewed and merged so I can roll into the v8 daily tests. That will help our test coverage as well convince the chromium API owners to gain more confidence and to give me the approval  sooner.

**Comment by FrankYFTang** [comment]
@gibson042

**Comment by rwaldron** [comment]
Just a heads up, both @leobalter and I are on vacation this week. I'll gladly review this first thing next week.

**Review by ryzokuken — APPROVED** [review]
Just a few comments, most of them nits.

**Comment by FrankYFTang** [comment]
PTAL

**Review by ryzokuken — DISMISSED** [review]
Changes LGTM, thanks for addressing the comments.

**Comment by ryzokuken** [comment]
@rwaldron would you have some time to review and eventually merge these soon? Thanks a ton.

**Review by rwaldron — APPROVED** [review]

**Pull request merged**